ipyvolume/pylab.py

```diff
@@ -1007,3 +979,47 @@ def lasso(data, other=None, fig=fig):
                     if selected:
                         scatter.selected = selected
     fig.on_lasso(lasso)
+    
+
+
+def make_triangles_lines(x, y, z, wrapx=False, wrapy=False):
+    """Transform rectangular regular grid into triangles
+    
+    :param x: {x2d}
+    :param y: {y2d}
+    :param z: {z2d}
+    :param bool wrapx: when True, the x direction is assumed to wrap, and polygons are drawn between the end end begin points
+    :param bool wrapy: simular for the y coordinate
+    :return: triangles and lines used to plot Mesh
+    """
+    assert len(x.shape) == 2, "Array x must be 2 dimensional."
+    assert len(y.shape) == 2, "Array y must be 2 dimensional."
+    assert len(z.shape) == 2, "Array z must be 2 dimensional."
+    assert x.shape == y.shape, "Arrays x and y must have same shape."
+    assert y.shape == z.shape, "Arrays y and z must have same shape."
+    
+    nx, ny = x.shape
+
+    mx = nx if wrapx else nx - 1
+    my = ny if wrapy else ny - 1
+    
+    i, j = np.mgrid[0:mx, 0:my]
```

Comment:
from this part on it's a bit difficult to follow, could you put in some inline comments to explain the idea? For instance give an example for nx,ny = (2,3) orso.

Reply:
I'll will look at some explanation this week-end, but the function does exactly the same things as your function but without using for loops wich are really slow in python

Reply:
I understand, and I don't think it's gonna be a problem to decypher it, but for maintainance etc it would be nice to explain it a bit.